cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRequestQueue.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.app.preview;
+
+import java.util.Optional;
+
+/**
+ * Interface designed for holding {@link PreviewRequest}s that are in WAITING state.
```

Comment:
Could you please add javadoc around what is the expected thread safety of this data structure implementation.

Reply:
Added doc to the implementation class.

Reply:
So thread safety is not a contract enforced by this interface?

Reply:
Should this interface enforces all implementations to be thread safe?

Reply:
Not sure I understand this comment. Did you mean adding `ThreadSafe` annotation to interface?

Reply:
Yes and clearly state that it is a requirement for the implementing class.

Reply:
Got it. done.